### ISSUE TITLE
MongoUnit demo 2 fails with Spring Boot 2.3 - No MongoDbFactory beans found.

### ISSUE THREAD

**Issue by choda** [mongounit/mongounit-demo2#2]
Hi,

Using the latest (2.3.1) Spring Boot version instead of the default 2.1.5 used in your example, tests fail with : 

```
  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::        (v2.3.1.RELEASE)

2020-07-07 10:10:17.246  INFO 3792 --- [           main] o.m.d.dao.mongo.MongoPersonDaoServiceIT  : Starting MongoPersonDaoServiceIT on DSI01801891PB with PID 3792 (started by MTARIN in C:\Users\mtarin\Dev\github\mongounit-demo2)
2020-07-07 10:10:17.248  INFO 3792 --- [           main] o.m.d.dao.mongo.MongoPersonDaoServiceIT  : No active profile set, falling back to default profiles: default
2020-07-07 10:10:17.681  INFO 3792 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data MongoDB repositories in DEFAULT mode.
2020-07-07 10:10:17.738  INFO 3792 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 53ms. Found 2 MongoDB repository interfaces.
2020-07-07 10:10:17.845  INFO 3792 --- [           main] o.m.config.MongoDbGuardConfiguration     : No MongoDbFactory beans found.
2020-07-07 10:15:09.862  INFO 1048 --- [           main] org.mongodb.driver.cluster               : Cluster created with settings {hosts=[localhost:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSelectionTimeout='30000 ms'}
2020-07-07 10:15:09.923  INFO 1048 --- [localhost:27017] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:1, serverValue:30}] to localhost:27017
2020-07-07 10:15:09.927  INFO 1048 --- [localhost:27017] org.mongodb.driver.cluster               : Monitor thread successfully connected to server with description ServerDescription{address=localhost:27017, type=STANDALONE, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=5, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=null, roundTripTimeNanos=1458952}
2020-07-07 10:15:10.400  INFO 1048 --- [           main] o.m.d.dao.mongo.MongoPersonDaoServiceIT  : Started MongoPersonDaoServiceIT in 2.13 seconds (JVM running for 3.311)
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.487 s <<< FAILURE! - in org.mongounit.demo.dao.mongo.MongoPersonDaoServiceIT
org.mongounit.demo.dao.mongo.MongoPersonDaoServiceIT  Time elapsed: 2.486 s  <<< ERROR!
org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.springframework.data.mongodb.MongoDbFactory' available
...
```

It still works fine with Spring Boot 2.2.8.

Any idea how to fix the configuration?

**Comment by ychaikin** [comment]
@choda thanks for reporting!

Hmm... Looks like `org.springframework.data.mongodb.MongoDbFactory` is no longer in the classpath. They must have replaced it or moved it to a different package... I'll investigate. Thanks!

**Comment by choda** [comment]
Yes, it was moved to the legacy driver in branch 4.
I managed to get mongounit to build and pass unit tests with Spring Boot 2.3/Mongo 4 and used it as a local dependency to the demo. But I still get the missing MongoDbFactory bean error.

Edit: actually, MongoDbFactory is now deprecated in favor of its superinterface MongoDatabaseFactory; the class moved to the legacy driver is MongoClientURI.

**Comment by ychaikin** [comment]
You mean you changed the code in mongounit and then built it? ... and then you used that newly built mongounit with the demo2 project and it still gave you that error?

**Comment by choda** [comment]
Exactly. But it was a quick&dirty attempt, I just added the legacy driver to the POM and replaced the deprecated stuff for mongounit to build without error.

**Comment by ychaikin** [comment]
The issue has been resolved with release 2.0.0 of mongoUnit: https://github.com/mongounit/mongounit/releases/tag/v2.0.0
